src/AWS.Deploy.Common/DeploymentManifest/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.IO;
+using Newtonsoft.Json;
+
+namespace AWS.Deploy.Common.DeploymentManifest
+{
+    public interface IDeploymentManifestEngine
+    {
+        string TargetApplicationFullPath { get; set; }
+
+        Task UpdateDeploymentManifestFile(string saveCdkDirectoryFullPath);
+    }
+
+    /// <summary>
+    /// This class contains the helper methods to update the deployment manifest file
+    /// that keeps track of the save CDK deployment projects.
+    /// </summary>
+    public class DeploymentManifestEngine : IDeploymentManifestEngine
+    {
+        private readonly IDirectoryManager _directoryManager;
+        private readonly IFileManager _fileManager;
+
+        public string TargetApplicationFullPath { get; set; } = string.Empty;
```

Comment:
I imagine you are assigning an empty string to get rid of error for initialization during construction. This should be a nullable string and must be asserted on null before it is being used in the methods. It allows avoiding developer error in case one forgets to set to a correct value.

Reply:
Fixed